src/coordinator.coffee

```diff
@@ -537,15 +573,16 @@ class Coordinator extends EventEmitter
         participantStartConcurrency = 10
         async.mapLimit options.only, participantStartConcurrency, (role, cb) =>
           componentName = graph.processes[role].component
-          @startParticipant role, componentName, cb
+          metadata = graph.processes[role].metadata or {}
+          @startParticipant role, componentName, metadata, cb
         , setupCallback
 
       setupConnections = (setupCallback) =>
         async.map graph.connections, (c, cb) =>
           if c.data
-            @addInitial c.tgt.process, c.tgt.port, c.data, cb
+            @addInitial c.tgt.process, c.tgt.port, c.data, c.metadata, cb
```

Comment:
Is `c.metadata` guaranteed to be an object? If it can be null/undefined, should normalize to empty object - either here or inside addInitial (maybe better).